File: app/src/main/java/com/softartdev/notecrypt/ui/splash/SplashActivity.java
package com.softartdev.notecrypt.ui.splash;

import android.content.Intent;
import android.content.res.Configuration;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.softartdev.notecrypt.ui.main.MainActivity;

import java.util.Locale;

public class SplashActivity extends AppCompatActivity implements SplashView {
    SplashPresenter mPresenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mPresenter = new SplashPresenter(this);
        mPresenter.setLocale();
        mPresenter.checkEncryption();
    }

    @Override
    public void navSignIn() {
        //TODO
    }

    @Override
    public void navMain() {
        Intent intent = new Intent(this, MainActivity.class);
        startActivity(intent);
        finish();
    }

    @Override
    public void onLocale(Locale locale) {
        Configuration configuration = getResources().getConfiguration();
        configuration.locale = locale;
        onConfigurationChanged(configuration);
    }

    @Override
    public void onConfigurationChanged(Configuration newConfig) {
        super.onConfigurationChanged(newConfig);
        getBaseContext().getResources().updateConfiguration(newConfig, getResources().getDisplayMetrics());
    }
}

File: app/src/main/java/com/softartdev/notecrypt/ui/splash/SplashActivity.kt
package com.softartdev.notecrypt.ui.splash

import android.content.Intent
import android.os.Bundle
import com.softartdev.notecrypt.ui.base.BaseActivity
import com.softartdev.notecrypt.ui.main.MainActivity
import com.softartdev.notecrypt.ui.signin.SignInActivity
import javax.inject.Inject

class SplashActivity : BaseActivity(), SplashView {
    @Inject lateinit var mPresenter: SplashPresenter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        activityComponent().inject(this)
        mPresenter.attachView(this)

        mPresenter.checkEncryption()
    }

    override fun navSignIn() {
        val intent = Intent(this, SignInActivity::class.java)
        startActivity(intent)
        finish()
    }

    override fun navMain() {
        val intent = Intent(this, MainActivity::class.java)
        startActivity(intent)
        finish()
    }

    override fun onDestroy() {
        mPresenter.detachView()
        super.onDestroy()
    }
}
